### PR TITLE
test: use testthat + codecov

### PR DESCRIPTION
**Pull request by elipousson**
These are all fairly basic changes to start addressing #5 and #6:

- test: use testthat + codecov
- test: add test for amend_timestamp
- docs: switch to Rmd for README + revise README + add package docs
- docs: add Eli Pousson to contributors
- refactor: add testthat, withr, and covr packages to Suggests
- style: style timestamp.R

I can set up a GitHub action with `use_github_action("test-coverage")` on main once this is merged.

### CODE REVIEW

**Comment by florisvdh** [comment]
Hi @elipousson, just to make sure, are you waiting for me to review, or am I right that I have nothing to do yet? No hurry as far as I'm concerned.

**Comment by elipousson** [comment]
Hello @florisvdh! Apologies for the delayed reply and for not responding to your email. I was thinking it may be beneficial to merge this pull request so you could take advantage of the automated checks from code codecov.io when reviewing a second pull requests with more of the additional functionality later on. If you'd rather wait to do it all at once, that is fine as well – I could just add additional features to this pull request and let you know when it is ready to merge.

I would like to tackle the updates in at least two PRs though – one to handle the new helper functions built with RSQLite (which could be this one if we don't merge it first) and a second to incorporate the extension access functions that rely on some structured data about the GeoPackage standard. Let me know what you think!

**Comment by florisvdh** [comment]
> I was thinking it may be beneficial to merge this pull request so you could take advantage of the automated checks from code codecov.io when reviewing a second pull requests with more of the additional functionality later on.

OK, then I'll review this PR in the course of coming days. Thanks! I was confused by it being still a draft (= unfinished) PR, that's all.

It's perfect to split up separate topics as several successive PRs.

**Comment by elipousson** [comment]
> I was confused by it being still a draft (= unfinished) PR, that's all.

Ah! That makes sense. Sorry for the mix-up.

File: README.Rmd [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# rgeopackage
+
+<!-- badges: start -->
+[![CRAN status](https://www.r-pkg.org/badges/version/rgeopackage)](https://CRAN.R-project.org/package=rgeopackage)
+[![License: GPL v3](https://img.shields.io/badge/License-GPLv3-blue.svg)](https://www.gnu.org/licenses/gpl-3.0)
+[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
+[![Codecov test coverage](https://codecov.io/gh/r-spatial/rgeopackage/branch/main/graph/badge.svg)](https://app.codecov.io/gh/r-spatial/rgeopackage?branch=main)
+<!-- badges: end -->
+
+The goal of rgeopackage is to support reading and writing metadata associated with GeoPackage (gpkg) files and assist in writing reproducible GeoPackage files.
```
Comment (florisvdh) [inline README.Rmd:25]:
```suggestion
The current goal of `rgeopackage` is to support reading and writing metadata associated with GeoPackage (gpkg) files and assist in writing reproducible GeoPackage files.
More generally, it aims to provide features not readily available in mainstream geospatial R packages, or perhaps uneasy to discover.
```

Comment (florisvdh) [reply README.Rmd:25]: Don't hesitate to update further or improve, this is an opinionated edit of course. You should feel free to take the lead!

File: README.Rmd [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Installation
+
+You can install the development version of rgeopackage from [GitHub](https://github.com/) with:
+
+``` r
+# install.packages("devtools")
+devtools::install_github("r-spatial/rgeopackage")
```
Comment (florisvdh) [inline README.Rmd:33]:
```suggestion
remotes::install_github("r-spatial/rgeopackage")
```

File: README.Rmd [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+[![License: GPL v3](https://img.shields.io/badge/License-GPLv3-blue.svg)](https://www.gnu.org/licenses/gpl-3.0)
+[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
+[![Codecov test coverage](https://codecov.io/gh/r-spatial/rgeopackage/branch/main/graph/badge.svg)](https://app.codecov.io/gh/r-spatial/rgeopackage?branch=main)
+<!-- badges: end -->
+
+The goal of rgeopackage is to support reading and writing metadata associated with GeoPackage (gpkg) files and assist in writing reproducible GeoPackage files.
+
+## Installation
+
+You can install the development version of rgeopackage from [GitHub](https://github.com/) with:
```
Comment (florisvdh) [inline README.Rmd:29]:
```suggestion
You can install the development version of `rgeopackage` from [GitHub](https://github.com/) with:
```

File: README.Rmd [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Overview
+
+### Updating file timestamps
+
+- `preset_timestamp()`: presets the file timestamp for usage by GDAL by setting the environment variable `OGR_CURRENT_DATE`.
+The timestamp is adopted by GDAL during the entire session, unless `unset_timestamp()` is called.
+- `amend_timestamp()`: overwrites timestamps in the `gpkg_contents` and `gpkg_metadata_reference` tables of an existing GeoPackage file.
+
+By default, GDAL sets timestamps corresponding to system time, so GeoPackages change when rewriting. Both functions accept a `Date` or `POSIXct` object and use standard formatting for GeoPackage files.
```
Comment (florisvdh) [inline README.Rmd:44]:
```suggestion
By default, GDAL sets timestamps corresponding to system time, so GeoPackages change when rewriting. Both functions accept a `Date` or `POSIXct` object and reformat to comply with the GeoPackage standard.
```

Comment (florisvdh) [reply README.Rmd:44]: I try to point to the strict GPKG standard more explicitly here. Feel free to rephrase as I'm no native English speaker.

File: README.Rmd [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+## Related R packages
+
+- The [sf package](https://r-spatial.github.io/sf/) allows users to set GDAL configuration options in `sf::st_write()` using the `config_options` argument ([see sf issue #1618](https://github.com/r-spatial/sf/issues/1618#issuecomment-811231056)) and in `sf::st_read()` using the `options` argument (see [sf issue 1157](https://github.com/r-spatial/sf/issues/1157)). sf also includes a set of GDAL functions (e.g. `sf::gdal_metadata()`) that are not intended to be called directly by the user but could be used to access and edit GeoPackage metadata.
+
+- The [vapour package](https://hypertidy.github.io/vapour/index.html) provides access to the basic read functions available in GDAL including GeoPackage contents and extension tables.
```
Comment (florisvdh) [inline README.Rmd:50]:
```suggestion
- The [`sf` package](https://r-spatial.github.io/sf/) allows users to set GDAL configuration options in `sf::st_write()` using the `config_options` argument ([see sf issue #1618](https://github.com/r-spatial/sf/issues/1618#issuecomment-811231056)) and in `sf::st_read()` using the `options` argument (see [sf issue 1157](https://github.com/r-spatial/sf/issues/1157)). These options only affect the specific read or write statement and do not persist during the session. `sf` also includes a set of GDAL functions (e.g. `sf::gdal_metadata()`) that are not intended to be called directly by the user but could be used to access and edit GeoPackage metadata.

- The [`vapour` package](https://hypertidy.github.io/vapour/index.html) provides access to the basic read functions available in GDAL including GeoPackage contents and extension tables.
```

**Review by florisvdh — APPROVED** [review]
Hi @elipousson, thanks so much for your effort and apologies for the delay. I intend to speed up my response in future PRs.

All seems fine, just some minor tweaks suggested in `README.Rmd`. Feel free to reword better where needed. Please re-knit since I didn't update the md file.

I generally advise to use a new line for each new sentence in (R)md files, since this makes the diffs smaller (`pandoc` rewraps in the output). But as I'm happy to pass the lead on this package, I'll just respect your personal preference :wink:.

Please mark PRs as ready for review from the moment a PR has everything you deem needed. If still a draft (unfinished), you can ask to review specifics of course, in between.

You're welcome to merge after applying suggestions you consider useful. I look forward to the GHA setup on `main`!

**Comment by elipousson** [comment]
Thanks, @florisvdh – your suggested changes all look reasonable to me. I'll go ahead and merge it, set up a GitHub action for testing, and work to get a more fully featured pull request ready over the next month or so.

**Comment by florisvdh** [comment]
Thanks Eli! Note, you still have to re-knit `README.Rmd` in order to update `README.md`.

**Comment by elipousson** [comment]
Thanks for the reminder, @florisvdh. Done!

**Pull request merged**